### PR TITLE
Close out CUDA arch migration

### PR DESCRIPTION
**Pull request by jakirkham**
At this point the logic between the CUDA `linux_64` and other `linux_*` builds is essentially the same. To simplify things, fold the CUDA logic for all `linux_*` builds into the global pinnings. Then close out the CUDA arch migrator. This should allow the arch migrator to enable CUDA builds as part of its migrations (without needing a second migrator PR to add CUDA arch builds).

<hr>

Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by jakirkham** [comment]
Tested removing the CUDA 11.2 arch migrator from a couple feedstocks one building natively on Travis and one using cross-compilation on Azure (`ucx` and `cupy`) and then re-rendering with these changes. There was no change to the feedstocks (as expected)

Going forward if a feedstock uses CUDA and an arch migration is requested, this will ensure only one migration PR is needed for both sets of changes (as opposed to 2: one CPU arch migration and one CUDA arch migration). This should cutdown on churn in feedstocks and make things easier to understand/work with

**Comment by jakirkham** [comment]
@conda-forge/core , this is ready for review! 🙏

**Comment by jakirkham** [comment]
Planning on merging tomorrow if no comments

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -36,14 +36,17 @@ m2w64_fortran_compiler:        # [win]
...
 cuda_compiler_version:
   - None
-  - 11.2                       # [(linux64 or win64) and os.environ.get("CF_CUDA_ENABLED", "False") == "True"]
+  - 11.2                       # [(linux or win64) and os.environ.get("CF_CUDA_ENABLED", "False") == "True"]
 cuda_compiler_version_min:
   - None                       # [osx]
   - 11.2                       # [linux or win64]
 
+arm_variant_type:              # [aarch64]
+  - sbsa                       # [aarch64]
```
Comment (kkraus14) [inline recipe/conda_build_config.yaml:48]: Should this be limited to where `CF_CUDA_ENABLED` is `TRUE` or does this key only get used in those situations anyway?

Does this impact any non-CUDA packages?

Comment (jakirkham) [reply recipe/conda_build_config.yaml:48]: So far this is mainly used with CUDA

Maybe it makes sense if there are CPU-only packages that do particular things on different ARM variants. Not aware of any such use cases today. Though given the nature of ARM, this doesn't seem too hard to imagine

**Review by kkraus14 — APPROVED** [review]

**Comment by jakirkham** [comment]
Thanks for taking a look Keith! 🙏

Happy to follow up on anything else as needed 🙂

**Pull request merged**